e2e/cypress/integration/channel_sidebar/category_collapsing_unread_filter_spec.js

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import users from '../../fixtures/users';
+
+import {testWithConfig} from '../../support/hooks';
+
+import {getRandomInt} from '../../utils';
+
+const sysadmin = users.sysadmin;
+
+describe('Channel sidebar', () => {
+    testWithConfig({
+        ServiceSettings: {
+            ExperimentalChannelSidebarOrganization: 'default_on',
+        },
+    });
+
+    before(() => {
+        cy.apiLogin('user-1');
+
+        cy.visit('/');
+    });
+
+    it('should display collapsed state when collapsed', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
```

Comment:
Should each test/it block needs to be tested against a new team? If possible, create a new team once for all the tests in this spec file. Same for other spec files.

Reply:
This is copied from my code. We wanted to start with a new team each time to avoid having one test depend on the state left over by the previous test. Maybe we should have some best practices listed around this?

Reply:
I totally agree on it. One thing that I'm worried is with the number of new teams for "user-1" which is a permanent/default test user. One downside is it might introduce flakiness to other tests like needing to do "scrollIntoView" since the expected team is outside the window.
Instead of user-1, one option is to create a new (throw-away) test user where it can be added into many teams as required; like fire and forget after the test.
But I'm fine as is. We'll look into this more during daily Cypress test.